src/views/lighttable.c

```diff
@@ -616,7 +626,71 @@ static int expose_filemanager(dt_view_t *self, cairo_t *cr, int32_t width, int32
         if(iir == 1 && row) continue;
 
         /* set mouse over id if pointer is in current row / col */
-        if(pi == col && pj == row)
+        if (lib->using_arrows)
+        {
+
+          if(before_mouse_over_id == -1)
+          {
+            // mouse has never been in filemanager area set mouse on first image and ignore this movement
+            before_mouse_over_id = query_ids[0];
+          }
+          if(before_mouse_over_id == id)
+          {
+            // I would like to jump from before_mouse_over_id to query_ids[idx]
+            int idx = current_image+lib->key_jump_offset-1;
+            int current_row = (int)((current_image-1)/iir);
+            int current_col = current_image%iir;
+
+            // detect if the current movement need some extra movement (page adjust)
+            if (current_row  == (int)(max_rows-1.5) && lib->key_jump_offset == iir)
+              // going DOWN from last row
+              move_view(lib,DOWN);
+            else if (current_row  == 0 && lib->key_jump_offset == iir*-1)
+            {
+              // going UP from first row
+              move_view(lib,UP);
+            }
+            else if (current_row == (int)(max_rows-1.5) && current_col ==  0 && lib->key_jump_offset == 1)
+              // going RIGHT from last visible
+              move_view(lib,DOWN);
+            else if (current_row == 0 && current_col ==  1 && lib->key_jump_offset == -1)
+              // going LEFT from first visible
+              move_view(lib,UP);
+            if (idx > -1 && idx < lib->collection_count && query_ids[idx])
```

Comment:
```
int *query_ids = (int *)calloc(max_rows * max_cols, sizeof(int));
```
```
idx < lib->collection_count
```

I'm pretty sure i can have selection bigger than what fits on one screen. (assuming that is what that code asserts)